applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -1726,6 +1726,37 @@ void CrBeamElement3D2N::FinalizeNonLinearIteration(const ProcessInfo& rCurrentPr
     SaveQuaternionParameters();
 }
 
+const Parameters CrBeamElement3D2N::GetSpecifications() const
+{
+    const Parameters specifications = Parameters(R"({
+        "time_integration"           : ["static","implicit","explicit"],
+        "framework"                  : "lagrangian",
+        "symmetric_lhs"              : true,
+        "positive_definite_lhs"      : true,
+        "output"                     : {
+            "gauss_point"            : ["MOMENT","FORCE","LOCAL_AXIS_1","LOCAL_AXIS_2","LOCAL_AXIS_3","INTEGRATION_COORDINATES"],
+            "nodal_historical"       : ["DISPLACEMENT","ROTATION","VELOCITY","ACCELERATION"],
+            "nodal_non_historical"   : [],
+            "entity"                 : []
+        },
+        "required_variables"         : ["DISPLACEMENT","ROTATION"],
+        "required_dofs"              : ["DISPLACEMENT_X","DISPLACEMENT_Y","DISPLACEMENT_Z","ROTATION_X","ROTATION_Y","ROTATION_Z"],
+        "flags_used"                 : [],
+        "compatible_geometries"      : ["Line3D2"],
+        "element_integrates_in_time" : false,
+        "compatible_constitutive_laws": {
+            "type"        : ["BeamConstitutiveLaw"],
+            "dimension"   : ["1D"],

```

Comment:
Still confused about this... (we can discuss later)

Reply:
IMO, the meaning is clear when dealing with (non-generalized) strains/stresses:
Uniaxial --> dimension = 1D --> size = 1 (sigma_xx)
Planar --> dimension = 2D --> size = 3 (sigma_xx, sigma_yy, tau_xy)
and the 3D case with size = 6.

The case of the 2D and 3D beams is different, as explained in my previous comment: they use generalized strains/stresses for the constitutive definition.